src/test/java/harness/TestCommand.java

```diff
@@ -34,19 +37,63 @@ public static Result runCommand(String... args) {
         new PrintStream(stdErr, true, StandardCharsets.UTF_8));
 
     // execute the command from the top-level Main class
+    System.out.println("COMMAND: " + String.join(" ", args));
     int exitCode = Main.runCommand(args);
 
+    // log the stdout and stderr to the console
+    String stdOutStr = stdOut.toString(StandardCharsets.UTF_8);
+    String stdErrStr = stdErr.toString(StandardCharsets.UTF_8);
+    System.out.println(stdOutStr);
+    if (!stdErrStr.isEmpty()) {
+      System.out.println("STDERR --------------");
+      System.out.println(stdErrStr);
+    }
+
     // return a result object with all the command outputs
-    return new Result(
-        exitCode, stdOut.toString(StandardCharsets.UTF_8), stdErr.toString(StandardCharsets.UTF_8));
+    return new Result(exitCode, stdOutStr, stdErrStr);
   }
 
-  /** Helper method to convert what's written to standard out into a Java object. */
+  /** Convert what's written to standard out into a Java object. */
   public static <T> T readObjectFromStdOut(Result cmd, Class<T> objectType)
       throws JsonProcessingException {
     return objectMapper.readValue(cmd.stdOut, objectType);
   }
 
+  /** Convert what's written to standard out into a Java object. */
+  public static <T> T readObjectFromStdOut(Result cmd, TypeReference<T> objectType)
+      throws JsonProcessingException {
+    return objectMapper.readValue(cmd.stdOut, objectType);
+  }
+
+  /** Helper method to run a command and check its exit code is 0=success. */
```

Comment:
Big time nit, but is anyone actually using the returned `Result`?  I looked for this in the test code and didn't see any, but might have missed, but if not...

With the three overloads it feels a little unexpected to me that the first would return the local `Result` while the other two throw it away and return something else entirely.  If the return is indeed not used anywhere, maybe consider `void` return, otherwise maybe a different name for the two that return parsed results?

Reply:
Good point. Renamed as follows:
- `runAndParseCommandExpectSuccess` (two methods) return the parsed result
- `runCommandExpectSuccess` (one method) has a `void` return
- `runCommand` (one method) returns the `Result` object. This one is used for checking error messages.